swagger.yaml

```diff
@@ -62,9 +150,31 @@ paths:
         500:
           description: "Failed to process the request due to an internal error"
   /datasets:
+    post:
+      tags:
+      - "Private user"
+      summary: "Create a dataset"
+      description: "Create a dataset provided by the ONS that can be filtered using the filter API"
+      parameters:
+      - $ref: '#/parameters/new_dataset'
+      produces:
+      - "application/json"
+      responses:
+        201:
+          description: "A json object containing a dataset which has been created"
+          schema:
+            $ref: '#/definitions/NewDatasetResponse'
+        400:
+          description: "Invalid request body"
+        401:
+          description: "Unauthorised to create/overwrite dataset"
+        403:
```

Comment:
A private user with authentication will never be forbidden to create a dataset (atm) as currently what do we go on to know whether this dataset already exists? What fields do we look for across all datasets?